src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1037,8 +1037,6 @@ async Task<int> ExecuteNonQuery(bool async, CancellationToken cancellationToken)
             using var reader = await ExecuteReader(CommandBehavior.Default, async, cancellationToken);
             while (async ? await reader.NextResultAsync(cancellationToken) : reader.NextResult()) ;
 
-            reader.Close();
```

Comment:
Did you need to remove this because it doesn't work? I'm asking because this usage pattern should absolutely work for users - it's even in the ADO.NET documentation.

Reply:
As per offline discussion, accessing RecordsAffected after calling Close is supported - but will create an additional allocation as the reader is unbound from the connector.

Reply:
(But only with multiplexing on)